common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -663,8 +684,13 @@ def get_problem_html(self, encapsulate=True):
         context = {
             'problem': content,
             'id': self.location.to_deprecated_string(),
-            'problem_is_timed': problem_is_timed,
+            'problem_is_timed': self.is_timed_problem(),
+            'problem_has_finished': self.closed() or self.is_submitted(),
+            'exceeded_time_limit': self.exceeded_time_limit() and not self.done,
+            'minutes_before_warning': self.minutes_before_warning,
             'start_time': self.time_started,
+            'seconds_left': seconds_left,
+            'minutes_allowed': self.minutes_allowed,
```

Comment:
To make a cleaner, more future-proof diff, can we group each of these keys together as a single block and define them as the very last keys in the dict?